processor/cumulativetodeltaprocessor/processor.go

```diff
@@ -16,35 +16,97 @@ package cumulativetodeltaprocessor
 
 import (
 	"context"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
+
+	awsmetrics "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/metrics"
 )
 
 type cumulativeToDeltaProcessor struct {
-	metrics []string
-	logger  *zap.Logger
+	metrics         map[string]bool
+	logger          *zap.Logger
+	deltaCalculator awsmetrics.MetricCalculator
 }
 
 func newCumulativeToDeltaProcessor(config *Config, logger *zap.Logger) *cumulativeToDeltaProcessor {
+	inputMetricSet := make(map[string]bool)
```

Comment:
```suggestion
	inputMetricSet := make(map[string]bool, len(config.Metrics))
```

Probably doesn't make a huge difference, but we can maybe save some reallocations by constructing the map with an appropriate size from the start.